addons/html_editor/static/tests/paste.test.js

```diff
@@ -766,12 +766,270 @@ describe("Simple html elements containing <br>", () => {
+        });
+        await testEditor({
+            contentBefore: "<p>mn[]</p>",
+            stepFunction: async (editor) => {
+                pasteHtml(editor, "<h1>abc</h1><h2>def</h2><h3>ghi</h3>");
+            },
+            contentAfter: "<p>mn</p><h1>abc</h1><h2>def</h2><h3>ghi[]</h3>",
+        });
+    });
+    test("should unwrap a node when pasting at end of same node", async () => {
+        await testEditor({
+            contentBefore: "<h1>mn[]</h1>",
+            stepFunction: async (editor) => {
+                pasteHtml(editor, "<h1>abc</h1>");
+            },
+            contentAfter: "<h1>mnabc[]</h1>",
+        });
+        await testEditor({
+            contentBefore: "<h1>mn[]</h1>",
+            stepFunction: async (editor) => {
+                pasteHtml(editor, "<h1>abc</h1><h2>def</h2>");
+            },
+            contentAfter: "<h1>mnabc</h1><h2>def[]</h2>",
+        });
+        await testEditor({
+            contentBefore: '<h1><font style="background-color: rgb(255, 0, 0);">mn[]</font></h1>',
+            stepFunction: async (editor) => {
+                pasteHtml(editor, "<h1>abc</h1><h1>def</h1><h1>ghi</h1>");
+            },
+            contentAfter:
+                '<h1><font style="background-color: rgb(255, 0, 0);">mnabc</font></h1><h1>def</h1><h1>ghi[]</h1>',
+        });
+    });
+    test("should paste all nodes as blockquote when pasting in blockquote", async () => {
+        await testEditor({
+            contentBefore: "<blockquote>[]<br></blockquote>",
+            stepFunction: async (editor) => {
+                pasteHtml(editor, "<h1>abc</h1><h2>def</h2><h3>ghi</h3>");
+            },
+            contentAfter:
+                "<blockquote>abc</blockquote><blockquote>def</blockquote><blockquote>ghi[]</blockquote>",
+        });
+        await testEditor({
+            contentBefore: "<blockquote>x[]</blockquote>",
+            stepFunction: async (editor) => {
+                pasteHtml(editor, "<h1>abc</h1><h2>def</h2><h3>ghi</h3>");
+            },
+            contentAfter:
+                "<blockquote>xabc</blockquote><blockquote>def</blockquote><blockquote>ghi[]</blockquote>",
+        });
+        await testEditor({
+            contentBefore: "<blockquote>[]x</blockquote>",
+            stepFunction: async (editor) => {
+                pasteHtml(editor, "<h1>abc</h1><h2>def</h2><h3>ghi</h3>");
+            },
+            contentAfter:
+                "<blockquote>abc</blockquote><blockquote>def</blockquote><blockquote>ghi[]x</blockquote>",
+        });
+        await testEditor({
+            contentBefore: "<blockquote>x[]y</blockquote>",
+            stepFunction: async (editor) => {
+                pasteHtml(editor, "<h1>abc</h1><h2>def</h2><h3>ghi</h3>");
+            },
+            contentAfter:
+                "<blockquote>xabc</blockquote><blockquote>def</blockquote><blockquote>ghi[]y</blockquote>",
+        });
+    });
```

Comment:
I'm not so sure about those. I know that's what PDE asked you to do in the chatter, but I think he misunderstood the question or wasn't clear enough what he meant in his reply. I think the expected result would be to paste the html as is inside the blockquote, like treating the blockquote as some kind of DIV. What do you think @lebl-odoo?